company-flow.el

```diff
@@ -0,0 +1,91 @@
+;;; company-flow.el --- Flow backend for company-mode  -*- lexical-binding: t -*-
+(require 'cl-lib)
+
+(defun company-flow-handle-signal (process _event)
+  (when (memq (process-status process) '(signal exit))
+    (let ((callback (process-get process 'company-flow-callback))
+          (prefix (process-get process 'company-flow-prefix)))
+      ;; (message "%S" (company-flow-parse-output (company-flow-get-output process)))
+      (funcall callback (->> process
+                             company-flow-get-output
+                             company-flow-parse-output
+                             (all-completions prefix))))))
+
+(defun company-flow-make-candidate (line)
+  (let* ((words (split-string line " "))
+         (text (car words))
+         (meta (mapconcat 'identity (cdr words) " ")))
+    (propertize text 'meta meta)))
+
+(defun company-flow-parse-output (output)
+  (mapcar 'company-flow-make-candidate
```

Comment:
Similarly, this one could be replaced with `re-search-forward` in the output buffer. `company-clang--parse-output` could be used as an example.